# Урок 1. Случайные события. Условная вероятность. Формула Байеса. Независимые испытания

In [2]:
from math import factorial

Функция вычисления кол-ва сочетаний n элементов по k элементов в каждом:

In [4]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

#### 1. Из колоды в 52 карты извлекаются случайным образом 4 карты. 

> a) Найти вероятность того, что все карты – крести.   
>б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

а) В первый раз из колоды можно извлечь $\displaystyle\frac{13}{52}$ крестей. Когда будем извлекать карты во второй, то и крестей, и общего количества карт станет меньше: $\displaystyle\frac{12}{51}$.  

При третьем извлечении: $\displaystyle\frac{11}{50}$.  

При четвертом: $\displaystyle\frac{10}{49}$.  

Так как событие зависимое (не возвращаем карту обратно в колоду), то:  

P(C) = $\displaystyle\frac{13}{52}$*$\displaystyle\frac{12}{51}$*$\displaystyle\frac{11}{50}$$\displaystyle\frac{10}{49}$ = 0.0026

б) Кол-во исходов, при которых можно достать один туз, извлекая 4 карты:  $C^{1}_{4}*C^{3}_{48}$ 

Тогда вероятность, что будет извлечено 2 туза: $C^{2}_{4}*C^{2}_{48}$   

3 туза:  $C^{3}_{4}*C^{1}_{48}$   

И, наконец, что все извлеченные карты окажутся тузами:  $C^{4}_{4}$  

Так как события совместные, то их следует сложить (логическое **ИЛИ**: **или** извлечем 1 туз, **или** 2 туза, **или** 3 туза, **или** 4):  

$C^{1}_{4}*C^{3}_{48}$ + $C^{2}_{4}*C^{2}_{48}$ + $C^{3}_{4}*C^{1}_{48}$ + $C^{4}_{4}$  

Воспользуемся формулой, чтобы упорстить вычисления:

In [14]:
n = sum([combinations(4, 1) * combinations(48, 3), combinations(4, 2) * combinations(48, 2), 
         combinations(4, 3) * combinations(48, 1),combinations(4, 4)])
n

76145

Выше было найдено благоприятное количество исходов. В то время как общее количество исходов:

In [15]:
N = combinations(52, 4)
N

270725

Следовательно, вероятность извлечь хотя бы один туз:

In [16]:
P = n / N
print(round(P, 4))

0.2813


Ответ:  а) 0.0026 = 0,26%                б) 0,2813 = 28,13%

****

#### 2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Сначала определим общее число возможных комбинаций 3-х элементов из 10 возможных:

$C^{3}_{10}$ = 10! / (3! * 7!) = 120 - это N

Тогда кол-во благоприятных исходов - 1, так как по условию дверь неоьходимо открыть с первого раза. Следовательно:

P = 1 / 120 = 0,0083

In [18]:
N = combinations(10, 3)
P = 1 / N
print(round(P, 4))

0.0083


Ответ: 0,00083 = 0,83%

____

#### 3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

Число благоприятных исходов - число комбинаций из 9 элементов по 3:

$C^{3}_{9}$ = 9! / (3! * 6!) = 84 [n]


Общее количество возможных комбинаций:

$C^{3}_{15}$ = 15! / (3! * 12!) = 455 [N]

Тогда вероятность выбрать 3 окрашенных шара:

P = n / N = 84 / 455 = 0,1846

In [21]:
n = combinations(9, 3)
N = combinations(15, 3)
P = n / N
print(round(P, 4))

0.1846


Ответ: 0.1846 = 18,46%

____

#### 4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

Кол-во благоприятных исходов - 1, так как $C^{2}_{2}$ = 2! / (2! * 0!) = 1 [n]

Общее кол-во возможных вариантов исхода событий: $C^{100}_{2}$ = 100! / (2! * 100!) = 4 950 [N]

Тогда вероятность выбрать 2 выигрышных билета из 100:

Р = n / N = 1 / 4 950 = 0.0002

In [26]:
P = 1 / combinations(100, 2)
print(round(P, 4))

0.0002


Ответ: 0.0002 = 0.02%

____

#### 5. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: 

> a) первым спортсменом   
б) вторым спортсменом   
в) третьим спортсменом.

В1 - первый спортсмен
В2 - второй спортсмен
В3 - третий спортсмен

А - попадание в мишень
P(A/B1) = 0.9
P(A/B2) = 0.8
P(A/B3) = 0.6

Формула Байеса:
$P(B/A) = \displaystyle\frac{P(B)*P(A/B)}{P(A)}$

Сперва посчитаем вероятность того, что выйдет первый/второй/третий спортсмен: $P(B) = \displaystyle\frac{1}{3}$

Затем посчитаем полную вероятность P(A) = P(B1) * P(A/B1) + P(B2) * P(A/B2) + P(B3) * P(A/B3) = $\displaystyle\frac{1}{3} * 0.9 + \displaystyle\frac{1}{3} * 0.8 + \displaystyle\frac{1}{3} * 0.6 = 0,7666$


Затем можно посчитать саму вероятность:

а) P(B1/A) =  (1 / 3 * 0.9) / 0.7666 = 0.3913

б) P(B2/A) =  (1 / 3 * 0.8) / 0.7666 = 0.3478

в) P(B3/A) =  (1 / 3 * 0.6) / 0.7666 = 0,2608

Ответ:
    
а) 0.3913 = 39,13%

б) 0.3478 = 34,78%

в) 0,2608 = 26,08%

_____